easybuild/easyblocks/d/dualsphysics.py

```diff
@@ -0,0 +1,151 @@
+##
+# Copyright 2013-2023 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing DualSPHysics, implemented as an easyblock
+
+@author: Jasper Grimm (University of York)
+"""
+import glob
+import os
+import stat
+
+from easybuild.easyblocks.generic.cmakemakecp import CMakeMakeCp
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.config import build_option
+from easybuild.tools.filetools import adjust_permissions
+from easybuild.tools.modules import get_software_root
+from easybuild.tools.run import run_cmd
+
+
+class EB_DualSPHysics(CMakeMakeCp):
+    """Support for building/installing DualSPHysics."""
+
+    @staticmethod
+    def extra_options():
+        """Extra easyconfig parameters for DualSPHysics."""
+        extra_vars = CMakeMakeCp.extra_options()
+
+        extra_vars['separate_build_dir'][0] = True
+
+        # files_to_copy is not mandatory here since we set it in the easyblock
+        extra_vars['files_to_copy'][2] = CUSTOM
+        return extra_vars
+
+    def __init__(self, *args, **kwargs):
+        """Initialize calss variables custom to DualSPHysics."""
+        super(EB_DualSPHysics, self).__init__(*args, **kwargs)
+
+        self.dsph_target = None
+        self.shortver = None
+
+    def prepare_step(self, *args, **kwargs):
+        """Determine name of binary that will be installed."""
+        super(EB_DualSPHysics, self).prepare_step(*args, **kwargs)
+
+        if get_software_root('CUDA'):
+            self.dsph_target = 'GPU'
+        else:
+            self.dsph_target = 'CPU'
+
+        self.shortver = '.'.join(self.version.split('.')[0:2])
+
+    def configure_step(self):
+        """Custom configure procedure for DualSPHysics."""
+        srcdir = os.path.join(self.cfg['start_dir'], 'src/source')
+        CMakeMakeCp.configure_step(self, srcdir=srcdir)
```

Comment:
Why not "super(EB_DUALSPHysics, self)...." as is used in other places (and is what we normally use)

Reply:
good point, somehow overlooked that
```suggestion
        super(EB_DUALSPHysics, self).configure_step(srcdir=srcdir)
```